File: LeetCode/jvm/src/main/java/com/sexton/leetcode/ReverseInteger.java
package com.sexton.leetcode;

public final class ReverseInteger {
    private ReverseInteger() {
        // Private constructor to stop instantiation
    }

    public static int reverseInteger(final int integer) {
        if (integer == 0) {
            return 0;
        }

        final String integerString;
        final String negativeValue;
        if (integer > 0) {
            integerString = String.valueOf(integer);
            negativeValue = "";
        } else {
            integerString = String.valueOf(integer).substring(1);
            negativeValue = "-";
        }

        final String reversedIntegerString = negativeValue + new StringBuilder(integerString).reverse();

        try {
            return Integer.parseInt(reversedIntegerString);
        } catch (final NumberFormatException ex) {
            // Number was too large to be converted to an integer.
            // Challenge specifies to just return 0
            return 0;
        }
    }
}

File: LeetCode/src/main/kotlin/com/sexton/leetcode/ReverseInteger.kt
package com.sexton.leetcode

fun Int.reverse(): Int {
    val reversedNumberString = this.toString().trimStart { it == '-' }.reversed().trimStart { it == '0' }
    val negative = if (this < 0) "-" else ""
    return try {
        if (reversedNumberString != "") "$negative$reversedNumberString".toInt() else 0
    } catch (e: NumberFormatException) {
        0
    }
}
